app/controllers/admin/api/cms/sections_controller.rb

```diff
@@ -101,9 +102,19 @@ def destroy
   end
 
   private
-    def find_section
-      @section = current_account.sections.find_by_id(params[:id]) || current_account.sections.find_by_system_name(params[:id])
 
-      raise ActiveRecord::RecordNotFound.new("Couldn't find CMS::Section with id or system_name=#{params[:id]}") if @section.nil?
-    end
+  def find_section(contents: false)
+    q = current_account.sections
+    q = q.includes(:pages, :files, :builtins, :children) if contents
```

Comment:
When we discussed we agreed to include IDs as content. But to get these IDs, one has to perform 4 extra queries per request. Is it worth it when one can get pages, files, builtins and children specifying parent? Returning IDs doesn't seem to save on API calls or db queries :(

Reply:
I think it's worth it because:

1. This feature and the search filter are complementary and both are going to be implemented eventually. So if in the future client we decide that the best way to get the whole contents tree is using the search filter and not this, that's ok.
2. The UI shows the contents when opening a particular section, so we should provide the same feature for the API.
3. If we remove this, there won't be a way to get the contents for one section in one single request, we'll need to request all pages for the lists of templates, sections and files filtered by parent_id in order to build the list of childs of one section, which is in the best of cases 3 requests and much more DB queries, since every request queries for the account, permissions, etc.
4. I don't expect a high traffic for this API anyway.

Reply:
Getting IDs is also not content. If one needs the content, then one can request the other items with filter. It is common that UI and API do not translate 1-1 with each other.

Reply:
Getting IDs is also not content. If one needs the content, then one can request the other items with filter. It is common that UI and API do not translate 1-1 with each other.

Reply:
Getting IDs is also not content. If one needs the content, then one can request the other items with filter. It is common that UI and API do not translate 1-1 with each other.

Reply:
I also think it's nice to have the contents of the section (the IDs of the elements inside of it). But my only concern is that if we add it, it might be more difficult to remove this in future.
But I'd vote for including it, and maybe if we see that it gets much traffic, we can optimize it by implementing some more efficient queries with joins.

Reply:
I think this is feature creep, when we don't know if something is useful or not but potentially inefficient.

In current state, it makes sense to me to include child objects summaries as well to the response, because we already get these from database anyway. So if client needs only summaries, then it will not have to perform further queries on children.

Reply:
So I actually have a pretty strong opinion here to remove this content. When we write a client, we can easily and painlessly add it. What is the benefit to add it now? We don't set the feature in stone and adding it later should not break existing clients. I see no drawbacks to leave it out for now.

Reply:
I agree with the latest statement.
We can have this one feature in our "TODO"  (or "maybe TODO") list for future, and add it at any later point.

Reply:
I removed this